app/uk/gov/hmrc/tariffclassificationfrontend/service/CasesService.scala

```diff
@@ -27,18 +31,41 @@ import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
 @Singleton
-class CasesService @Inject()(connector: BindingTariffClassificationConnector) {
-
-  def releaseCase(c: Case, queue: Queue, operator: Operator)(implicit hc: HeaderCarrier): Future[Case] = {
-    val eventualCase: Future[Case] = connector.updateCase(c.copy(status = CaseStatus.OPEN, queueId = Some(queue.id)))
-    eventualCase.onSuccess({
-      case updated =>
-        connector.createEvent(updated, NewEventRequest(CaseStatusChange(c.status, updated.status), operator.id))
-          .onFailure({
-            case throwable: Throwable => Logger.error(s"Could not create Release Case event for case [${c.reference}]", throwable)
-          })
-    })
-    eventualCase
+class CasesService @Inject()(appConfig: AppConfig, auditService: AuditService, connector: BindingTariffClassificationConnector) {
+
+  private def addEvent(original: Case, updated: Case, operator: Operator)(implicit hc: HeaderCarrier): Future[Unit] = {
+    val event = NewEventRequest(CaseStatusChange(original.status, updated.status), operator.id)
+    connector.createEvent(updated, event)
+      .recover({
+        case throwable: Throwable =>
+          Logger.error(s"Could not create Event for case [${original.reference}] with payload [$event]", throwable)
+      })
+      .map(_ => Unit)
+  }
+
+  def releaseCase(original: Case, queue: Queue, operator: Operator)(implicit hc: HeaderCarrier): Future[Case] = {
+    for {
+      updated <- connector.updateCase(original.copy(status = CaseStatus.OPEN, queueId = Some(queue.id)))
+      _ <- addEvent(original, updated, operator)
+      _ <- auditService.auditCaseReleased(updated)
```

Comment:
once you convert the type from `Future[Unit]` to `Unit`, this line will become:
```
_ = auditService.auditCaseReleased(updated)
```

Reply:
sorted